src/NUnitFramework/framework/Attributes/RandomAttribute.cs

```diff
@@ -258,14 +283,36 @@ public override IEnumerable GetData(IParameterInfo parameter)
 
                 _randomizer = Randomizer.GetRandomizer(parameter.ParameterInfo);
 
+                if (Distinct && _inRange && !CanBeDistinct(_min, _max, _count))
+                    throw new NUnitException($"Random values can't be distinct. Method: \"{parameter.Method}\"; Parameter: \"{parameter.ParameterInfo.Name}\"");
```

Comment:
What if we didn't throw but just ignored distinct in this case. I don't like ignoring user errors in general, but this one seems somewhat harmless. What do other @nunit/framework-team folks think?

If we do want a message, then 
1. It should be more user friendly, like "Cannot generate 5 distinct values for X because the range only contains 4 values" or "Cannot generate 5 distinct values from 1 to 3 for X."

2. We generally use standard exceptions in situations like this. See `Guard.ArgumentValid`. 

3. `NUnitException` is only used as the base class of our own exceptions and to wrap exceptions in certain cases.